commit 0f61633a6014d6bce330e39b7a5ab0f88da60de8
Author: Raphael Scholle <raphael@openhdfpv.org>
Date:   Fri Mar 15 23:56:31 2024 +0100

    modify utils

diff --git a/lib/utils.sh b/lib/utils.sh
index bdb1bcd..2b5dd86 100644
--- a/lib/utils.sh
+++ b/lib/utils.sh
@@ -199,6 +199,31 @@ prepare_source() {
             git reset --hard FETCH_HEAD
             git clean -ffd
             git switch --detach $origin/$BSP_BRANCH
+
+	    #openhd-update-wifi-card-drivers
+            echo "________OpenHD Part_________"
+            cd drivers/net/wireless/
+            rm -Rf rtl8812au
+            rm -Rf rtl88x2bu
+            git clone https://github.com/openhd/rtl8812au/
+            git clone https://github.com/openhd/rtl88x2bu/
+            git clone https://github.com/openhd/rtl88x2bu/ -b initial52-1 rtl8852bu
+            cd rtl8812au
+            sed -i 's/CONFIG_PLATFORM_I386_PC = y/CONFIG_PLATFORM_I386_PC = n/' Makefile
+            sed -i 's/CONFIG_PLATFORM_ARM64_RPI = n/CONFIG_PLATFORM_ARM64_RPI = y/' Makefile
+            cd ..
+            cd rtl88x2bu
+            sed -i 's/CONFIG_PLATFORM_I386_PC = y/CONFIG_PLATFORM_I386_PC = n/' Makefile
+            sed -i 's/CONFIG_PLATFORM_ARM64_RPI = n/CONFIG_PLATFORM_ARM64_RPI = y/' Makefile
+            cd ..
+            cd rtl8852bu
+            sed -i 's/aarch64\.l/arm64/' Makefile
+            cd ../../../../
+            patch -p1 -i scripts/patches/openhd_drm.patch
+            patch -p1 -i scripts/patches/openhd_wifi.patch
+            patch -p1 -i scripts/patches/openhd_usb.patch
+            #patch -p1 -i scripts/patches/openhd_keyboard.patch
+            echo "________OpenHD END_________"
         fi
 
         if $LONG_VERSION
