commit b0988eb0223230690e413a6250eaceba1226cdee
Author: Raphael <68374617+raphaelscholle@users.noreply.github.com>
Date:   Mon Jun 17 02:09:52 2024 +0200

    i2c-patch

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
index 62b0e763db2a..9b71b7abd50f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
@@ -323,6 +323,11 @@ regulator-state-mem {
 	};
 };
 
+&i2c7 {
+	status = "okay";
+	pinctrl-0 = <&i2c7m3_xfer>;
+};
+
 &dmc {
 	center-supply = <&vdd_ddr_s0>;
 	mem-supply = <&vdd_log_s0>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
index ebed3da7a556..e85316332fb0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
@@ -211,6 +211,11 @@ regulator-state-mem {
 	};
 };
 
+&i2c8 {
+	status = "okay";
+	pinctrl-0 = <&i2c8m2_xfer>;
+};
+
 &i2c2 {
 	status = "okay";
 	pinctrl-names = "default";
