commit d3a053a5d47dddc8b4f7e68b284415e322a53945
Author: Raphael <68374617+raphaelscholle@users.noreply.github.com>
Date:   Sun Apr 21 19:55:26 2024 +0200

    enable uart

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-aic8800ds2.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-aic8800ds2.dts
index e11ba6772eea..f209ba11a86a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-aic8800ds2.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-aic8800ds2.dts
@@ -58,6 +58,12 @@ &uart1 {
 	pinctrl-0 = <&uart1m0_xfer &uart1m0_ctsn>;
 };
 
+&uart2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2m0_xfer>;
+};
+
 &sdmmc1 {
 	supports-sdio;
 	bus-width = <4>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts
index 0abc6e60db50..1b617414ff5b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts
@@ -62,6 +62,12 @@ bluetooth {
 	};
 };
 
+&uart2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2m0_xfer>;
+};
+
 &sdmmc1 {
 	supports-sdio;
 	bus-width = <4>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero3.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero3.dts
index a8a02c1572c2..8375fa10df42 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero3.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero3.dts
@@ -537,6 +537,12 @@ bluetooth {
 	};
 };
 
+&uart2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2m0_xfer>;
+};
+
 &gmac1 {
 	phy-mode = "rgmii";
 	clock_in_out = "output";
